### ISSUE TITLE
parameters file creates an offset that results inability to complete evalutaion

### ISSUE THREAD

**Issue by probinso** [GaloisInc/ppaml-cp5#7]
Given the directory trees
```script
/tmp/umd-output/
├── pcfgla.grammar
├── pcfgla.lexicon
└── pcfgla.states

./demo/data/
├── sample_gold.txt
├── sample_pred.txt
├── sample_test.txt
└── sample_train.txt
```

and the evaluation envocation `eval.sh    /tmp/umd-output/    ./demo/data/    /tmp/umd-results`

while executing [line 75](https://github.com/GaloisInc/ppaml-cp5/blob/master/eval.sh#L75 ) the program errors out and prints  
```script
1 : Length unmatch (29|28)
4 : Length unmatch (8|7)
5 : Length unmatch (21|20)
8 : Length unmatch (68|69)
18 : Length unmatch (21|20)
21 : Length unmatch (22|21)
26 : Length unmatch (14|13)
28 : Length unmatch (19|18)
30 : Length unmatch (23|24)
32 : Length unmatch (20|19)
33 : Length unmatch (35|34)
34 : Length unmatch (56|57)
```
to stderr, then terminates without producing or saving the f1 score.

If we rewrite the line to be `$evalb ${gold_file} ${parsed_file} > ${eval_file} 
` by removing the parameters file, then this behavior looks to produce an f1 score.

Can I get some feedback on this behavior. Do we need to use the parameters file?

**Comment by hhexiy** [comment]
Hi,

The parameter file is needed. My run seems to be fine though. Could you
send (or paste the first few lines) of your pred and gold files?

**Comment by probinso** [comment]
The pred and gold files that I'm using are the ones in the repository.
```script
demo/
├── data
│     ├── sample_gold.txt
│     ├── sample_pred.txt
│     ├── sample_test.txt
│     └── sample_train.txt
├── eng_MLE.gr
├── eng_MLE.gr.grammar
├── eng_MLE.gr.info
├── eng_MLE.gr.lexicon
├── eng_MLE.gr.states
├── eng_MLE.read.gr
├── Makefile
└── README
```

The crash that happens is very subtle, likely because of the redirect to `${eval_file}`. Could you check to make sure that your run produces the subsequent `${f1_file}` produced on [line 80](https://github.com/GaloisInc/ppaml-cp5/blob/master/eval.sh#L80) .

**Comment by hhexiy** [comment]
Yes. I do get the f1_file. The error message means the test tree and the
gold tree has different number of leaves (words). Could you check one line
in the pred and gold file?

**Comment by probinso** [comment]
Alright,

when i train the model using the invocation from the provided `run.sh` the resulting lexicon only contains unknown words, and the result is that the program dumps out 
`unknown signature for word ...`  errors for every word

Can you please check to see that `run.sh` as provided in the base directory is being called as expected?

**Comment by hhexiy** [comment]
Hi,

This is because the training set is too small (100 sentences), and all
words appearing fewer than 20 times are replaced with their signatures
(UNK-*). However, the UNK words in the test file are collected by
thresholding counts from the entire training corpus, so it's very likely
that a common word becomes a rare word in the grammar trained on 100
sentences, hence the unknown word error.

Maybe we should put a pre-trained baseline model in the repo?

He

**Comment by probinso** [comment]
I'll run an evaluation end-to-end today over the data hosted on kitware and get back to you. This may end up being a non-problem.
